### PR TITLE
fix: display skip reason in new ui

### PR DESCRIPTION
**Pull request by shadowusr**
### What's done:
- Skip reason is now displayed as part of preview in tree view in new ui
- Skip reason is displayed in meta when possible

Demo report: https://nda.ya.ru/t/Ba_4zPC979ttwt

### CODE REVIEW

File: lib/static/new-ui/components/MetaInfo/index.tsx [hunk 1]
```diff
@@ -109,6 +110,23 @@ function MetaInfoInternal(props: MetaInfoInternalProps): ReactNode {
         });
     }
 
+    const shouldAddSkipReason = Boolean(!metaInfoItemsWithResolvedUrls.find(item => item.label === 'muteReason'));
```
Comment (DudaGod) [inline lib/static/new-ui/components/MetaInfo/index.tsx:113]: I would use a separate variable for this expression - `Boolean(!metaInfoItemsWithResolvedUrls.find(item => item.label === 'muteReason'))`. For example - `hasMuteReason`

File: lib/static/new-ui/components/MetaInfo/index.tsx [hunk 2]
```diff
@@ -109,6 +110,23 @@ function MetaInfoInternal(props: MetaInfoInternalProps): ReactNode {
...
+                label: 'skipReason',
+                content: reason
+            });
+        } else if (!Array.isArray(reason) && typeof reason.props.children === 'string' && typeof reason.props.href === 'string') {
+            metaInfoItemsWithResolvedUrls.push({
+                label: 'skipReason',
+                content: reason.props.children,
+                url: reason.props.href
+            });
+        }
```
Comment (DudaGod) [inline lib/static/new-ui/components/MetaInfo/index.tsx:127]: can `reason` be something else? I mean is there a condition under which the value does not fall into any of the described conditions (`if` and `else if`).

File: lib/static/new-ui/components/MetaInfo/index.tsx [hunk 3]
```diff
@@ -109,6 +110,23 @@ function MetaInfoInternal(props: MetaInfoInternalProps): ReactNode {
...
 
+    const shouldAddSkipReason = Boolean(!metaInfoItemsWithResolvedUrls.find(item => item.label === 'muteReason'));
+    if (result.skipReason && shouldAddSkipReason) {
+        const reason = Parser(result.skipReason);
+        if (typeof reason === 'string') {
+            metaInfoItemsWithResolvedUrls.push({
+                label: 'skipReason',
+                content: reason
+            });
+        } else if (!Array.isArray(reason) && typeof reason.props.children === 'string' && typeof reason.props.href === 'string') {
```
Comment (DudaGod) [inline lib/static/new-ui/components/MetaInfo/index.tsx:121]: Reason can be an array?

File: lib/static/new-ui/features/suites/components/TreeViewItemSubtitle/index.tsx [hunk 4]
```diff
@@ -16,7 +17,11 @@ interface TreeViewItemSubtitleProps {
 }
 
 export function TreeViewItemSubtitle(props: TreeViewItemSubtitleProps): ReactNode {
-    if (props.item.images?.length) {
+    if (props.item.skipReason) {
+        return <div className={styles.skipReasonContainer}>
+            <div className={styles.skipReason}>Skipped ⋅ {Parser(props.item.skipReason)}</div>
```
Comment (DudaGod) [inline lib/static/new-ui/features/suites/components/TreeViewItemSubtitle/index.tsx:22]: ` ⋅ ` - why you choose a dot as a separator?  But in IU it looks quite harmonious. Maybe you use it in some other components?

**Review by DudaGod — APPROVED** [review]
🔥

File: lib/static/new-ui/features/suites/components/TreeViewItemSubtitle/index.tsx (hunk 4 above)
Comment (shadowusr) [reply lib/static/new-ui/features/suites/components/TreeViewItemSubtitle/index.tsx:22]: I think it looks nice as a separator and it's not the only place we use dot. The other places are hint toast, and other variants of subtitle in tree view.

File: lib/static/new-ui/components/MetaInfo/index.tsx (hunk 1 above)
Comment (shadowusr) [reply lib/static/new-ui/components/MetaInfo/index.tsx:113]: We discussed this with sipayrt and decided that it's not very good idea to include such specific logic related to `muteReason`, so i got rid of altogether

File: lib/static/new-ui/components/MetaInfo/index.tsx (hunk 2 above)
Comment (shadowusr) [reply lib/static/new-ui/components/MetaInfo/index.tsx:127]: This logic is no longer here.

File: lib/static/new-ui/components/MetaInfo/index.tsx (hunk 3 above)
Comment (shadowusr) [reply lib/static/new-ui/components/MetaInfo/index.tsx:121]: This is no longer here, got rid of parsing html entirely.

**Pull request merged**